algebird-core/src/main/scala/com/twitter/algebird/CountMinSketch.scala

```diff
@@ -448,6 +450,50 @@ case class CMSItem[K: Ordering](item: K, override val params: CMSParams[K]) exte
 }
 
 /**
+ * Used for holding a small set of elements.
+ */
+case class CMSItems[K: Ordering](items: List[K], override val totalCount: Long, override val params: CMSParams[K]) extends CMS[K](params) {
+
+  override def +(x: K, count: Long): CMS[K] = CMSItems(x :: items, totalCount + 1, params).toCMS
+
+  override def ++(other: CMS[K]): CMS[K] = {
+    other match {
+      case other: CMSZero[_] => this
+      case other: CMSItem[K] => this + other.item
+      case other: CMSItems[K] => {
+        val (larger, smaller) =
+          if (totalCount > other.totalCount)
+            (this, other)
+          else
+            (other, this)
+
+        CMSItems(
+          larger.items.reverse ::: (smaller.items),
```

Comment:
>  (2) don't we want the smaller one on the left so we push the fewer items onto the bigger?

The code is correctly pushing the smaller list onto the larger list:

```scala
scala> val larger = List(1,2,3,4,5,6,7,8,9)
larger: List[Int] = List(1, 2, 3, 4, 5, 6, 7, 8, 9)

scala> val smaller = List(10,20,30)
smaller: List[Int] = List(10, 20, 30)

scala> larger.reverse_:::(smaller)
res4: List[Int] = List(30, 20, 10, 1, 2, 3, 4, 5, 6, 7, 8, 9)
```

> (1) don't we want the method reverse_::: which combines these two to save cost?

And I was surprised to learn that apparently a List's `reverse_:::` is faster than `++` or `:::` because it leverages structural sharing (its loop does [`::`](https://github.com/scala/scala/blob/v2.11.4/src/library/scala/collection/immutable/List.scala#L110)) whereas e.g. [`:::`](https://github.com/scala/scala/blob/v2.11.4/src/library/scala/collection/immutable/List.scala#L124) creates a `ListBuffer` via [`++=`](https://github.com/scala/scala/blob/v2.11.4/src/library/scala/collection/mutable/ListBuffer.scala#L181) and [`prependToList`](https://github.com/scala/scala/blob/v2.11.4/src/library/scala/collection/mutable/ListBuffer.scala#L316).  I didn't expect that, though I haven't yet verified my understanding with a quick caliper benchmark. :-)